packages/clerk-js/src/core/resources/Organization.ts

```diff
@@ -0,0 +1,141 @@
+import type {
+  GetMembersParams,
+  MembershipRole,
+  OrganizationInvitationJSON,
+  OrganizationJSON,
+  OrganizationMembershipJSON,
+  OrganizationResource,
+} from '@clerk/types';
+import { unixEpochToDate } from 'utils/date';
+
+import {
+  BaseResource,
+  OrganizationInvitation,
+  OrganizationMembership,
+} from './internal';
+
+export class Organization extends BaseResource implements OrganizationResource {
+  id!: string;
+  name!: string;
+  role!: MembershipRole;
+  instanceId!: string;
+  createdBy!: string;
+  createdAt!: Date;
+  updatedAt!: Date;
+
+  constructor(data: OrganizationJSON) {
+    super();
+    this.fromJSON(data);
+  }
+
+  static async create(name: string): Promise<OrganizationResource> {
+    const json = (
+      await BaseResource._fetch<OrganizationJSON>({
+        path: '/organizations',
+        method: 'POST',
+        body: { name } as any,
+      })
+    )?.response as unknown as OrganizationJSON;
+
+    return new Organization(json);
+  }
+
+  static retrieve(
+    getOrganizationParams?: GetOrganizationParams,
+  ): Promise<Organization[]> {
+    return this.clerk
+      .getFapiClient()
+      .request<OrganizationJSON[]>({
+        method: 'GET',
+        path: '/me/organizations',
+        search: getOrganizationParams as any,
+      })
+      .then(res => {
+        const organizationsJSON = res.payload
+          ?.response as unknown as OrganizationJSON[];
```

Comment:
❓ Just asking, why is the type assertion needed here?

I'd expect that we wouldn't need it because of the generic param in line 48: 
```
 .request<OrganizationJSON[]>({
```
Is this type broken?